src/tests/ftest/io/nvme_io_verification.py

```diff
@@ -0,0 +1,113 @@
+        """Jira ID: DAOS-2649.
+
+        Test Description:
+            Test will run IOR with non standard transfer sizes for different
+            set of pool sizes. Purpose is to verify io transaction to scm and
+            nvme for different pool sizes under different situations.
+
+        Use Cases:
+            (1) Running IOR with different set of transfer size where first
+            transfer size is < 4096 and then > 4096. Verify that data goes to
+            scm if transfer size < 4096 and then it goes to nvme if transfer
+            size is > 4096.
+            (2) Repeat the case(1) with maximum nvme pool size that can be
+            created.
+            (3) Running IOR with different set of transfer size where the
+            transfer size is > 4096 throughout. Verify that data goes to nvme
+            as transfer size is > 4096.
+            (4) Repeat the case(3) with maximum nvme pool size that can be
+            created.
+        :avocado: tags=all,daosio,full_regression,hw,nvme_io_verification
+        """
+        # Test params
+        tests = self.params.get("ior_sequence", '/run/ior/*')
+        processes = self.params.get("np", '/run/ior/*')
+        transfer_size = self.params.get("tsize", '/run/ior/transfersize/*/')
+        block_size = self.ior_cmd.block_size.value
+
+        # Loop for every IOR object type
+        for ior_param in tests:
+            # Create and connect to a pool
+            # pylint: disable=attribute-defined-outside-init
+            self.pool = TestPool(self.context, self.log)
+            self.pool.get_params(self)
+
+            # update pool sizes
+            self.pool.scm_size.update(ior_param[0])
+            self.pool.nvme_size.update(ior_param[1])
+
+            # Create a pool
+            self.pool.create()
+
+            # get pool info
+            self.pool.get_info()
+
+            for tsize in transfer_size:
+                # Get the current pool sizes
+                size_before_ior = self.pool.info
+
+                # Run ior with the parameters specified for this pass
+                self.ior_cmd.transfer_size.update(tsize)
+                # if transfer size is less thank 1K
+                # update block size to 32K to keep it small
+                if tsize <= 1000:
+                    self.ior_cmd.block_size.update(32000)
+                else:
+                    self.ior_cmd.block_size.update(block_size)
+                self.ior_cmd.set_daos_params(self.server_group, self.pool)
+                self.run_ior(self.get_job_manager_command(), processes)
+
+                # Verify IOR consumed the expected amount ofrom the pool
+                self.verify_pool_size(size_before_ior, self.processes)
+
+            try:
+                if self.pool:
+                    # pylint: disable=attribute-defined-outside-init
+                    self.pool.destroy(1)
+                    self.pool = None
+            except DaosApiError as error:
```

Comment:
The `TestPool.destroy()` method catches the `DaosApiError` exception and instead returns an `avocado.core.exceptions.TestFail` exception.  Here you want to catch the `TestFail` exception.

Reply:
Will do